src/hotspot/share/gc/g1/g1BarrierSet.cpp

```diff
@@ -120,6 +120,10 @@ void G1BarrierSet::invalidate(MemRegion mr) {
   volatile jbyte* byte = _card_table->byte_for(mr.start());
   jbyte* last_byte = _card_table->byte_for(mr.last());
   Thread* thr = Thread::current();
```

Comment:
is G1BarrierSet::write_ref_field_post_slow not used? perhaps add assert !G1BarrierSimple

Reply:
similarly for G1BarrierSetRuntime::write_ref_field_post_entry

Reply:
Assertion added